FIAT/hdiv_trace.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (C) 2016 Thomas H. Gibson
+#
+# This file is part of FIAT.
+#
+# FIAT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# FIAT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with FIAT. If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from FIAT.discontinuous_lagrange import DiscontinuousLagrange
+from FIAT.reference_element import ufc_simplex
+from FIAT import FiniteElement
+from FIAT.polynomial_set import mis
+
+
+class TraceError(Exception):
+    """Exception caused by tabulating a trace element on the interior of a cell,
+    or the gradient of a trace element."""
+
+    def __init__(self, msg, values, keys):
+
+        super(TraceError, self).__init__(msg)
+        self.msg = msg
+        self.zeros = values
+        self.D = keys
+
+
+class TraceHDiv(FiniteElement):
+    """Class implementing the trace of hdiv elements on general simplices."""
+
+    def __init__(self, cell, polyDegree):
+
+        spaceDim = cell.get_spatial_dimension()
```

Comment:
I know flake8 doesn't report for this, but this breaks Python conventions. Camel case should only be used for class names, variable and function names should use underscores.